### PR TITLE
Add Random Generator and unit tests

### LINKED ISSUE #4: Create bingo generator class

### LINKED ISSUE #11: Get next number from random generator

### PR DESCRIPTION
**Pull request by T-Syme**
- When a RandomGenerator is created, a full list of possible numbers is created, then shuffled randomly so there cannot be repeated numbers.
- The method getNext() returns the next number from the shuffled list
- Unit tests were written for getNext()

### CODE REVIEW

**Comment by T-Syme** [comment]
Running the tests locally using Maven test
image

File: src/test/java/RandomGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Test
+    void testGetNext() {
+        try {
+            for (int i = minVal; i <= maxVal; i++) {
+                int val = randomGenerator.getNext();
+                assertTrue(val >= minVal);
+                assertTrue(val <= maxVal);
+            }
+        } catch (Exception e) {
+            fail();
```
Comment (hanagustyn) [inline src/test/java/RandomGeneratorTest.java]: Adding a string message to the fail() method will help debug any errors. This also applies for line 45.

**Review by hanagustyn — CHANGES_REQUESTED** [review]
The code looks good. Please see my comment before merging the code.

I think a good next step for the random generator class would be to create a random letter generator as well.

**Comment by T-Syme** [comment]
I'm not sure I understand what the random letter generator would be used for. The bingo boards that I've seen seem to use numbers. Was there another purpose for it?
image

**Comment by hanagustyn** [comment]
I think Bingo usually has a letter and a number associated to it. For example, when someone calls out a slot they would say B12 which means you can cover that square if you have 12 in the B column.

It might be better though to represent those letters with numbers from 0 to 4, and then each number can be associated to a letter. That way, it will work better with an array for the game board.

**Review by hanagustyn — APPROVED** [review]
Looks good! Feel free to merge.

**Comment by T-Syme** [comment]
Huh, I didn't realize that was a thing. 
What do you think if I created an object and returned that? It could look something like this:
**RandomValue**
- randomValue.getColumn() --> number 0-4 (for b-i-n-g-o)
- randomValue.getNumber() --> number 1-75 (for bingo numbers)

**Comment by hanagustyn** [comment]
Sounds good to me. But this would be in addition to the random generator you created, right? Because I think that is still useful for generating the game board.

**Comment by T-Syme** [comment]
That's true. I can add another method that returns an array of 24 random values (5x5 with 1 free) for generating boards

**Comment by hanagustyn** [comment]
I actually have implemented some of that for the game board class already. I think I can just make an instance of RandomGenerator and then get 24 values from it.

**Comment by hanagustyn** [comment]
But I could also just implement that in the GameBoard class itself. Maybe that makes more sense?

**Comment by T-Syme** [comment]
Yes, I think it would be fine to call getNext() 24 times. I think it would be nice to generate all the random numbers from the same class

**Comment by hanagustyn** [comment]
Ok, sounds good. I'll do that.

**Comment by T-Syme** [comment]
Added the RandomValues as mentioned above
Using separate lists and iterators to make sure they don't clash when generating boards vs moves

**Review by hanagustyn — APPROVED** [review]
Looks good!

**Pull request merged**